### PR TITLE
Update prettier and eslint setup

### PR DESCRIPTION
**Pull request by DerrykBoyd**
## What issue is this referencing?

This brings the prettier and eslint setups up to date with how the tools are recommended to be used.

Remove prettier rules form eslint and run prettier separately as per https://prettier.io/docs/en/integrating-with-linters.html

We should add "npm run format-check" to this issue template

Include js files in linting (was only vue files before)

Fix the formatting on files that were not covered in prev settings.

## Have you run the linting command in the `README.md`?

-   [x] yes! `npm run lint-check`

## Have you taken a look at our [contributing guidelines](https://github.com/devedmonton/DES-Website/blob/main/.github/CONTRIBUTING.md)?

-   [x] yes!

## My node version matches the one suggested when running `nvm use`?

-   [x] yes!

### CODE REVIEW

File: store/README.md [hunk 1]
```diff
@@ -93,57 +93,57 @@ For more detailed instructions, including [Nuxt configuration](https://pinia.vue
...
-  },
-})
+    setup() {
+        const main = useMainStore();
+
+        // extract specific store properties
+        const { counter, doubleCounter } = storeToRefs(main);
+
+        return {
+            // gives access to the whole store in the template
```
Comment (MandyMeindersma) [inline store/README.md:139]: so this uses tabs

File: .devcontainer/devcontainer.json [hunk 2]
```diff
@@ -1,29 +1,27 @@
...
 // https://github.com/microsoft/vscode-dev-containers/tree/v0.134.1/containers/javascript-node
 {
-	"name": "devedmonton.com",
-	"build": {
-		"dockerfile": "Dockerfile",
-		// Update 'VARIANT' to pick a Node version: 10, 12, 14
-		"args": { "VARIANT": "14" }
-	},
+  "name": "devedmonton.com",
+  "build": {
```
Comment (MandyMeindersma) [inline .devcontainer/devcontainer.json:5]: and this uses spaces?

**Review by MandyMeindersma — COMMENTED** [review]
Why does it change whether we have 2-4 spaces depending on the file type? Is that normal? I thought they were always project wide decisions?

**Comment by DerrykBoyd** [comment]
> > Why does it change whether we have 2-4 spaces depending on the file type? Is that normal? I thought they were always project wide decisions?
> 

On another look, we have a `.editorconfig` file that overrides some of the prettier settings. That's where this difference is coming from. So just to confirm

- these are all spaces, not tabs (confirmed in my ediotor)
- Spaces are set to 4 for code and 2 for json in the editorconfig file

I would be up for just removing the editorconfig file and defaulting to the default prettier. That would require a format commit that would change pretty much all files.

**Comment by MandyMeindersma** [comment]
I feel like that's too much effort ahhahah

I am good to merge as is :P

**Comment by DerrykBoyd** [comment]
> I feel like that's too much effort ahhahah
> 
> I am good to merge as is :P 

Ha yeah, I'm easy. Not much effort but let's just leave it as is. Will keep the git history cleaner.

**Review by MandyMeindersma — APPROVED** [review]

**Comment by MandyMeindersma** [comment]
so we should add `npm run format-check` to the readme and the pr template? they are both in the repo

**Comment by olimpiuus** [comment]
@DerrykBoyd, that's a fantastic commit! Could we also consider adding something like the following?

`"lint-fix-and-format": "eslint . --ext .js,.vue --fix && prettier --write ."`

This script could be quite useful for automatically fixing linting issues and formatting the code. However, I'm unsure about its appropriateness. What do you think?

**Comment by DerrykBoyd** [comment]
> 

Yeah sure, I don't see any issue with that. Not something I'd use since my editor is now formatting / fixing on save. But if it's useful for others there's no harm.

Would suggest

```json
  "lint-fix-and-format": "npm run lint-fix && npm run format",
```

So we don't have duplicated commands to maintain

**Pull request merged**